### PR TITLE
Add option to disable calling compinit. 

### PR DESCRIPTION
**Pull request by mfarrugi**
Add variable `ZSH_DISABLE_COMPINIT`. When set to true, stops omzsh from invoking `compinit`.

Particularly useful when Omzsh is being used as a plugin by zgen, antigen, etc., and the user wants to call `compinit` later.

### CODE REVIEW

File: oh-my-zsh.sh [hunk 1]
```diff
@@ -64,19 +64,21 @@ if [ -z "$ZSH_COMPDUMP" ]; then
...
+      if ! compaudit &>/dev/null; then
+        # This function resides in the "lib/compfix.zsh" script sourced above.
+        handle_completion_insecurities
+      # Else, enable and cache completions to the desired file.
+      else
+        compinit -d "${ZSH_COMPDUMP}"
+      fi
   else
-    compinit -d "${ZSH_COMPDUMP}"
+      compinit -i -d "${ZSH_COMPDUMP}"
```
Comment (sachin21) [inline oh-my-zsh.sh]: I think 2 indents is better.

File: oh-my-zsh.sh [hunk 2]
```diff
@@ -64,19 +64,21 @@ if [ -z "$ZSH_COMPDUMP" ]; then
...
-if [[ $ZSH_DISABLE_COMPFIX != true ]]; then
-  # If completion insecurities exist, warn the user without enabling completions.
-  if ! compaudit &>/dev/null; then
-    # This function resides in the "lib/compfix.zsh" script sourced above.
-    handle_completion_insecurities
-  # Else, enable and cache completions to the desired file.
+
+if [[ $ZSH_DISABLE_COMPINIT != true ]]; then
+  if [[ $ZSH_DISABLE_COMPFIX != true ]]; then
+      # If completion insecurities exist, warn the user without enabling completions.
```
Comment (sachin21) [inline oh-my-zsh.sh]: I think 2 indents is so better, too

Comment (apjanke) [inline oh-my-zsh.sh]: Yeah, 2-space indents are [standard for Oh My Zsh](https://github.com/robbyrussell/oh-my-zsh/wiki/Coding-style-guide#indentation), and match the rest of this file.

**Comment by mcornella** [comment]
`compinit` needs to be called because many plugins rely on it to call `compdef`.

**Comment by mfarrugi** [comment]
@mcornella compinit  needs to be called once, at the end. Sometimes ohmyzsh is used as a library, instead of a package manager (eg. with zgen) and we would like for ohmyzsh to not call compinit a second time, as it can be perceptibly slow.

It's fine for this not to be upstreamed, as downstream package managers can be more explict about piecing out ohmyzsh.

For anyone else that stumbles across this, https://github.com/mfarrugi/.dots/blob/master/zsh/_zsh_plugins.zsh#L16 is my config for zplug which doesn't end up calling compinit twice.

**Pull request closed without merging**